### PR TITLE
Feature/miner with presealed

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
### Description of the Change
Add miner presealed sectors option.
Rework paths to boost::fs::paths.

### CODE REVIEW

File: core/common/tarutil.hpp [hunk 1]
```diff
@@ -16,9 +17,15 @@ namespace fc::common {
   outcome::result<void> zipTar(const std::string &input_path,
                                const std::string &output_path);
```
Comment (turuslan) [inline core/common/tarutil.hpp:19]:
```suggestion
  outcome::result<void> zipTar(const std::string &input_path,
                               const std::string &output_path);

```
`boost::filesystem::path` contains `std::string`, can remove string overload

File: core/sector_storage/stores/impl/remote_store.cpp [hunk 2]
```diff
@@ -382,8 +382,7 @@ namespace fc::sector_storage::stores {
     ec.clear();
 
     if (content_type == "application/x-tar") {
-      auto result =
-          fc::common::extractTar(temp_file_path.string(), output_path);
+      auto result = fc::common::extractTar(temp_file_path, output_path);
```
Comment (turuslan) [inline core/sector_storage/stores/impl/remote_store.cpp:385]:
```suggestion
      auto result = common::extractTar(temp_file_path, output_path);
```

**Review by turuslan — APPROVED** [review]

**Review by ortyomka — APPROVED** [review]

File: core/common/file.cpp [hunk 3]
```diff
@@ -33,9 +33,10 @@ namespace fc::common {
     return {};
   }
 
-  outcome::result<void> writeFile(std::string_view path, BytesIn input) {
-    // TODO: mkdir
-    std::ofstream file{path.data(), std::ios::binary};
+  outcome::result<void> writeFile(const boost::filesystem::path &path,
+                                  BytesIn input) {
+    boost::filesystem::create_directories(path.parent_path());
```
Comment (turuslan) [inline core/common/file.cpp:38]:
```suggestion
    if (path.has_parent_path()) {
      boost::filesystem::create_directories(path.parent_path());
    }
```

File: core/common/file.cpp [hunk 4]
```diff
@@ -33,9 +33,10 @@ namespace fc::common {
...
+    std::ofstream file{path.c_str(), std::ios::binary};
     if (file.good()) {
       file.write(span::bytestr(input.data()), input.size());
```
Comment (turuslan) [inline core/common/file.cpp:41]:
```suggestion
    if (file.write(span::bytestr(input.data()), input.size()).good()) {
```

**Pull request merged**